### DISCUSSION TITLE
fetchMore in FirestoreQueryBuilder,

### DISCUSSION THREAD

**Discussion by ablbol** [firebase/flutterfire#8206] [Q&A]
I would like to clarify one thing about fetchMore in FirestoreQueryBuilder.  So let's say my page size is 10 documents and currently one page of 10 documents is loaded. When I call fetchMore() does this function load the next 10 documents and append them to the previous 10  documents or does it load the previous 10 documents and the next 10 documents?
Thanks for your help..

**Comment by rrousselGit** [comment]
fetchMore subscribes to both the previous page and the new page

This is necessary since items can be removed or inserted in the previous page, impacting the next page

**Comment by ablbol** [comment]
Thanks for your answer but my question is does fetchMore() only fetches the new pages then appends it to the previous pages or does it fetch all previous pages and the next page.  So if currently I have 10 pages and I call fetchMore(), does this function fetch only page 11 or does it fetch 11 pages?

**Reply by rrousselGit** [reply]
It's subscribing to the 11 pages. I don't know the internals of firestore well enough to know whether it is re-fetching the existing items though

cc @russellwheatley @Salakar

**Reply by kulture-rob-snider** [reply]
Looking at `_fetchNextPage`, it ultimately does:
```
final expectedDocsCount = (_pageCount + 1) * widget.pageSize + 1;
final query = widget.query.limit(expectedDocsCount);
```

So like @rrousselGit said, it is fetching everything existing, plus the new page.

I think the question you are trying to ask (because I have the same) is does the refetching count as a billable read.

E.g. page size 10: does loading page 1, then page 2, then page 3 count as 30 billable reads (10 + 10 + 10) or 60 billable reads (10 + 20 + 30).
